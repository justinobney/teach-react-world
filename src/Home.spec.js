import React from 'react';
import {create} from 'react-test-renderer';
import Home from './Home';

describe('Component: Home', () => {
  it('basic render', () => {
    const tree = create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('alter state by triggering click callback', () => {
    const component = create(<Home />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    /*
    * - div.container
    *   - div.starter-template
    *     - h1
    *     - button <-- onClick
    *     - hr
    *     - Alert
    */
    tree.children[0].children[1].props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
