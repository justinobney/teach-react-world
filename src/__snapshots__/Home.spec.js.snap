exports[`Component: Home alter state by triggering click callback 1`] = `
<div
  className="container">
  <div
    className="starter-template">
    <h1>
      Lets Learn React
    </h1>
    <button
      className="btn btn-default"
      onClick={[Function]}>
      Toggle Alert Type
    </button>
    <hr />
    <div
      className="alert alert-info"
      role="alert">
      <strong>
        Step 2
         - 
      </strong>
      Presentational Component
    </div>
  </div>
</div>
`;

exports[`Component: Home alter state by triggering click callback 2`] = `
<div
  className="container">
  <div
    className="starter-template">
    <h1>
      Lets Learn React
    </h1>
    <button
      className="btn btn-default"
      onClick={[Function]}>
      Toggle Alert Type
    </button>
    <hr />
    <div
      className="alert alert-danger"
      role="alert">
      <strong>
        Step 2
         - 
      </strong>
      Presentational Component
    </div>
  </div>
</div>
`;

exports[`Component: Home basic render 1`] = `
<div
  className="container">
  <div
    className="starter-template">
    <h1>
      Lets Learn React
    </h1>
    <button
      className="btn btn-default"
      onClick={[Function]}>
      Toggle Alert Type
    </button>
    <hr />
    <div
      className="alert alert-info"
      role="alert">
      <strong>
        Step 2
         - 
      </strong>
      Presentational Component
    </div>
  </div>
</div>
`;
